include/swift/AST/DiagnosticsSema.def

```diff
@@ -2814,7 +2814,7 @@ WARNING(expression_unused_result_call,none,
 WARNING(expression_unused_result_operator,none,
         "result of operator %0 is unused", (DeclName))
 WARNING(expression_unused_result_unknown, none,
-        "result of call is unused, but produces %0", (Type))
+        "result of call to function returning %0 is unused", (Type))
```

Comment:
Nit: You may not necessarily be calling a function if this is emitted, you may be calling a closure.

I think you can probably delete this and just use `expression_unused_result_call` if it's all the same.

Reply:
@CodaFi I will update it.